src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,207 @@
+                handleFindTask(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("deadline")) {
+                handleDeadlineTask(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("event")) {
+                handleEventTask(input, listOfTasks);
+                return;
+            }
+
+            if (input.equals("list")) {
+                handleList(listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("delete")) {
+                handleDelete(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("checkdue")) {
+                handleCheckDue(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("mark")) {
+                handleMark(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("unmark")) {
+                handleUnmark(input, listOfTasks);
+                return;
+            }
+
+            throw new DukeExceptions("");
+        } catch (DukeExceptions DE) {
+            System.out.println(DE.toString());
+        }
+    }
+
+    public void handleFindTask(String input, TaskList listOfTasks) throws DukeExceptions {
+        try {
+            String[] checkForKeywordArr = input.split(" ");
+            if (checkForKeywordArr.length < 2) {
+                throw new DukeExceptions("find");
+            }
+            String keyword = input.substring(4);
+            listOfTasks.findTasks(keyword);
+        } catch (DukeExceptions DE) {
+            System.out.println(DE.toString());
+        }
+    }
+
+    /**
+     * Function to handle ToDo tasks from the user input.
+     * @param input String representation of the user's input.
+     * @param listOfTasks The list of tasks which has been loaded into the Duke chatbot instance.
+     * @throws DukeExceptions if name of the task is not properly given by user.
+     */
+    public void handleToDoTask(String input, TaskList listOfTasks) throws DukeExceptions {
+        try {
+            String useForInit = input.substring(4);
+            Task toAdd = new ToDo(useForInit);
+            listOfTasks.addTask(toAdd);
+        } catch (DukeExceptions DE) {
+            System.out.println(DE.toString());
+        }
+    }
+
+    /**
+     * Function to handle Deadlines tasks from the user's input.
+     * @param input String representation of the user's input.
+     * @param listOfTasks The list of tasks which has been loaded into the Duke chatbot instance.
+     * @throws DukeExceptions if date of task is wrongly given or if name of task is not properly given by user.
+     */
+    public void handleDeadlineTask(String input, TaskList listOfTasks) throws DukeExceptions {
+        try {
+            String useForInit = input.substring(8);
+            Task toAdd = new Deadlines(useForInit);
```

Comment:
Perhaps instead of useForInit, the variable name can describe the name eg. deadlineDescription